onedal/linear_model/incremental_linear_model.py

```diff
@@ -0,0 +1,150 @@
+        Algorithm used for computation on oneDAL side
+    """
+
+    def __init__(self, fit_intercept=True, copy_X=False, algorithm="norm_eq"):
+        module = self._get_backend("linear_model", "regression")
+        super().__init__(fit_intercept=fit_intercept, copy_X=copy_X, algorithm=algorithm)
+        self._partial_result = module.partial_train_result()
+
+    def _reset(self):
+        module = self._get_backend("linear_model", "regression")
+        self._partial_result = module.partial_train_result()
+
+    def partial_fit(self, X, y, queue=None):
+        """
+        Computes partial data for basic statistics
+        from data batch X and saves it to `_partial_result`.
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data batch, where `n_samples` is the number of samples
+            in the batch, and `n_features` is the number of features.
+
+        y: array-like of shape (n_samples,) or (n_samples, n_targets) in
+            case of multiple targets
+            Responses for training data.
+
+        queue : dpctl.SyclQueue
+            If not None, use this queue for computations.
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        module = self._get_backend("linear_model", "regression")
+
+        if not hasattr(self, "_policy"):
+            self._policy = self._get_policy(queue, X)
+
+        if not hasattr(self, "_dtype"):
+            self._dtype = get_dtype(X)
+            self._params = self._get_onedal_params(self._dtype)
+
+        if self._dtype not in [np.float32, np.float64]:
+            self._dtype = np.float64
+
+        X = X.astype(self._dtype, copy=self.copy_X)
+        y = y.astype(dtype=self._dtype)
+        self._y_ndim_1 = y.ndim == 1
+
+        X, y = _check_X_y(X, y, force_all_finite=False, accept_2d_y=True)
+
+        self.n_features_in_ = _num_features(X, fallback_1d=True)
+
+        X, y = _convert_to_supported(self._policy, X, y)
+        X_table, y_table = to_table(X, y)
+        hparams = get_hyperparameters("linear_regression", "train")
+        if hparams is not None and not hparams.is_default:
+            self._partial_result = module.partial_train(
+                self._policy,
+                self._params,
+                hparams.backend,
+                self._partial_result,
+                X_table,
+                y_table,
+            )
+        else:
+            self._partial_result = module.partial_train(
+                self._policy, self._params, self._partial_result, X_table, y_table
+            )
+
+    def finalize_fit(self, queue=None):
+        """
+        Finalizes linear regression computation and obtains coefficients
+        from the current `_partial_result`.
+
+        Parameters
+        ----------
+        queue : dpctl.SyclQueue
+            Not used here, added for API conformance
+
```

Comment:
do we only have fit() implemented for GPU in OneDAL c++?

Reply:
no, we have for all devices. in finalize_fit queue is not necessary because the policy is taken from previous call of partial_fit (currently there should be data in method's args to get the policy, that's why it is impossible to get policy directly in finalize_fit).